-- INFO - 19:00:42--03-15-2025:
-- -- LOG STARTED ======================================================

-- DEBUG - 19:00:42--03-15-2025:
-- -- Starting new HTTPS connection (1): stats.nba.com:443

-- DEBUG - 19:00:42--03-15-2025:
-- -- https://stats.nba.com:443 "GET /stats/leaguegamefinder?Conference=&DateFrom=12%2F14%2F2024&DateTo=12%2F14%2F2024&Division=&DraftNumber=&DraftRound=&DraftTeamID=&DraftYear=&EqAST=&EqBLK=&EqDD=&EqDREB=&EqFG3A=&EqFG3M=&EqFG3_PCT=&EqFGA=&EqFGM=&EqFG_PCT=&EqFTA=&EqFTM=&EqFT_PCT=&EqMINUTES=&EqOREB=&EqPF=&EqPTS=&EqREB=&EqSTL=&EqTD=&EqTOV=&GameID=&GtAST=&GtBLK=&GtDD=&GtDREB=&GtFG3A=&GtFG3M=&GtFG3_PCT=&GtFGA=&GtFGM=&GtFG_PCT=&GtFTA=&GtFTM=&GtFT_PCT=&GtMINUTES=&GtOREB=&GtPF=&GtPTS=&GtREB=&GtSTL=&GtTD=&GtTOV=&LeagueID=00&Location=&LtAST=&LtBLK=&LtDD=&LtDREB=&LtFG3A=&LtFG3M=&LtFG3_PCT=&LtFGA=&LtFGM=&LtFG_PCT=&LtFTA=&LtFTM=&LtFT_PCT=&LtMINUTES=&LtOREB=&LtPF=&LtPTS=&LtREB=&LtSTL=&LtTD=&LtTOV=&Outcome=&PORound=&PlayerID=&PlayerOrTeam=P&RookieYear=&Season=&SeasonSegment=&SeasonType=&StarterBench=&TeamID=&VsConference=&VsDivision=&VsTeamID=&YearsExperience= HTTP/1.1" 200 2192

-- INFO - 19:00:42--03-15-2025:
-- -- Fetched 37 game logs from 12/14/2024

-- INFO - 19:00:42--03-15-2025:
-- -- Cleaning and transforming 37 game logs for insert

-- ERROR - 19:00:42--03-15-2025:
-- -- Error fetching games for 12/14/2024 after 0 try(ies)
Traceback (most recent call last):
  File "D:\programming_projects\nba_etl\src\data\fetch.py", line 27, in get_game_logs
    return CleanGameLogs(game_logs_df)
  File "D:\programming_projects\nba_etl\src\data\clean.py", line 15, in __init__
    game_df = self.game(self.raw_df)
  File "D:\programming_projects\nba_etl\src\data\clean.py", line 85, in game
    game_df = drop_away_matchups(df[['game_id', 'season_id', 'game_date', 'matchup']].drop_duplicates())
  File "D:\programming_projects\nba_etl\src\data\clean.py", line 52, in drop_away_matchups
    if df['game_id'] in ['0022401229', '0022401230']:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\programming_projects\nba_etl\.venv\Lib\site-packages\pandas\core\generic.py", line 1577, in __nonzero__
    raise ValueError(
    ...<2 lines>...
    )
ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
-- DEBUG - 19:00:52--03-15-2025:
-- -- https://stats.nba.com:443 "GET /stats/leaguegamefinder?Conference=&DateFrom=12%2F14%2F2024&DateTo=12%2F14%2F2024&Division=&DraftNumber=&DraftRound=&DraftTeamID=&DraftYear=&EqAST=&EqBLK=&EqDD=&EqDREB=&EqFG3A=&EqFG3M=&EqFG3_PCT=&EqFGA=&EqFGM=&EqFG_PCT=&EqFTA=&EqFTM=&EqFT_PCT=&EqMINUTES=&EqOREB=&EqPF=&EqPTS=&EqREB=&EqSTL=&EqTD=&EqTOV=&GameID=&GtAST=&GtBLK=&GtDD=&GtDREB=&GtFG3A=&GtFG3M=&GtFG3_PCT=&GtFGA=&GtFGM=&GtFG_PCT=&GtFTA=&GtFTM=&GtFT_PCT=&GtMINUTES=&GtOREB=&GtPF=&GtPTS=&GtREB=&GtSTL=&GtTD=&GtTOV=&LeagueID=00&Location=&LtAST=&LtBLK=&LtDD=&LtDREB=&LtFG3A=&LtFG3M=&LtFG3_PCT=&LtFGA=&LtFGM=&LtFG_PCT=&LtFTA=&LtFTM=&LtFT_PCT=&LtMINUTES=&LtOREB=&LtPF=&LtPTS=&LtREB=&LtSTL=&LtTD=&LtTOV=&Outcome=&PORound=&PlayerID=&PlayerOrTeam=P&RookieYear=&Season=&SeasonSegment=&SeasonType=&StarterBench=&TeamID=&VsConference=&VsDivision=&VsTeamID=&YearsExperience= HTTP/1.1" 200 2192

-- INFO - 19:00:52--03-15-2025:
-- -- Fetched 37 game logs from 12/14/2024

-- INFO - 19:00:52--03-15-2025:
-- -- Cleaning and transforming 37 game logs for insert

-- ERROR - 19:00:52--03-15-2025:
-- -- Error fetching games for 12/14/2024 after 1 try(ies)
Traceback (most recent call last):
  File "D:\programming_projects\nba_etl\src\data\fetch.py", line 27, in get_game_logs
    return CleanGameLogs(game_logs_df)
  File "D:\programming_projects\nba_etl\src\data\clean.py", line 15, in __init__
    game_df = self.game(self.raw_df)
  File "D:\programming_projects\nba_etl\src\data\clean.py", line 85, in game
    game_df = drop_away_matchups(df[['game_id', 'season_id', 'game_date', 'matchup']].drop_duplicates())
  File "D:\programming_projects\nba_etl\src\data\clean.py", line 52, in drop_away_matchups
    if df['game_id'] in ['0022401229', '0022401230']:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\programming_projects\nba_etl\.venv\Lib\site-packages\pandas\core\generic.py", line 1577, in __nonzero__
    raise ValueError(
    ...<2 lines>...
    )
ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
